src/authoring/components/workspace/ai-settings.tsx

```diff
@@ -0,0 +1,260 @@
+
+import React, { useMemo } from "react";
+import { useForm, SubmitHandler, useFieldArray } from "react-hook-form";
+
+import { IWorkspaceConfigComponentProps } from "./common";
+import { AIEvaluation, Summarizer } from "src/authoring/types";
+
+interface CommentTag {
+  label: string;
+  display: string;
+}
+
+interface AISettingsFormInputs {
+  tagPrompt: string;
+  commentTags: CommentTag[];
+  aiEvaluation: AIEvaluation;
+  aiPrompt: {
+    systemPrompt: string;
+    mainPrompt: string;
+    categorizationDescription: string;
+    keyIndicatorsPrompt: string;
+    discussionPrompt: string;
+    summarizer: Summarizer;
+  },
+  aiTileAvailable: boolean;
+  showIdeasButton: boolean;
+}
+
+const AISettings: React.FC<IWorkspaceConfigComponentProps> = ({ unitConfig, setUnitConfig, saveState }) => {
+  const settings: AISettingsFormInputs = useMemo(() => {
+    const { tagPrompt, aiEvaluation, commentTags: rawCommentTags, aiPrompt } = unitConfig?.config || {};
+
+    const commentTags: CommentTag[] = !rawCommentTags
+      ? []
+      : Object.entries(rawCommentTags).map(([label, display]) => ({ label, display }));
+
+    // TODO: figure out how to determine these
+    const aiTileAvailable = false;
+    const showIdeasButton = false;
+
+    return {
+      tagPrompt: tagPrompt ?? "",
+      commentTags: commentTags ?? [],
+      aiEvaluation: aiEvaluation as AIEvaluation ?? "categorize-design",
+      aiPrompt: {
+        systemPrompt: aiPrompt?.systemPrompt ?? "",
+        mainPrompt: aiPrompt?.mainPrompt ?? "",
+        categorizationDescription: aiPrompt?.categorizationDescription ?? "",
+        keyIndicatorsPrompt: aiPrompt?.keyIndicatorsPrompt ?? "",
+        discussionPrompt: aiPrompt?.discussionPrompt ?? "",
+        summarizer: (aiPrompt?.summarizer ?? "image") as Summarizer,
+      },
+      aiTileAvailable,
+      showIdeasButton
+    };
+  }, [unitConfig]);
+
+  const { handleSubmit, register, control, formState: { errors } } = useForm<AISettingsFormInputs>({
+    defaultValues: settings
+  });
+  const commentTagsFieldArray = useFieldArray({ control, name: "commentTags" });
+
+  const onSubmit: SubmitHandler<AISettingsFormInputs> = (data) => {
+    setUnitConfig(draft => {
+      const config = draft?.config;
+      if (config) {
+        config.tagPrompt = data.tagPrompt;
+        config.aiEvaluation = data.aiEvaluation;
+        config.commentTags = data.commentTags.reduce((obj, tag) => {
+          if (tag.label && tag.display) {
+            obj[tag.label] = tag.display;
+          }
+          return obj;
+        }, {} as Record<string, string>);
+
+        const aiPrompt = config.aiPrompt || {};
```

Comment:
Perhaps you want to do:

```suggestion
        const aiPrompt = config.aiPrompt ||= {};
```
Otherwise I don't think these changes will get saved into config if there isn't already an `aiPrompt` property.

Reply:
Good catch.

Reply:
Fixed in https://github.com/concord-consortium/collaborative-learning/pull/2666/commits/5b19a0705a0f269d742caced12bd1299027d7f6f.